src/coord/src/catalog.rs

```diff
@@ -307,6 +336,21 @@ impl CatalogItem {
                 i.create_sql = do_rewrite(i.create_sql)?;
                 Ok(CatalogItem::Index(i))
             }
+            CatalogItem::Type(i) => match i {
+                Type::Map {
+                    create_sql,
+                    plan_cx,
+                    oid,
+                    key_ids,
+                    value_ids,
+                } => Ok(CatalogItem::Type(Type::Map {
+                    create_sql: do_rewrite(create_sql.clone())?,
+                    plan_cx: plan_cx.clone(),
+                    oid: *oid,
+                    key_ids: key_ids.clone(),
+                    value_ids: value_ids.clone(),
+                })),
+            },
```

Comment:
The current `ALTER RENAME` logic is not equipped to do this sort of rewriting, unfortunately. I think we need to just outright ban renaming types.